screens/SendSetupScreen.js

```diff
@@ -341,16 +455,33 @@ const SendSetupScreen = ({
                 fadeIn={false}
                 onRead={e => {
                   const qrData = e.data;
-                  const { address, amount } = parseQr(qrData);
-
-                  // If there's an amount, set the type to crypto
-                  amount && setAmountType("crypto");
 
-                  address && setToAddress(address);
-                  amount && setSendAmount(amount);
+                  const {
+                    address,
+                    amount,
+                    tokenAmount,
+                    label,
+                    tokenId,
+                    symbol
+                  } = parseQr(qrData, tokensById);
 
                   setErrors([]);
                   setQrOpen(false);
+
+                  // coinName and imageSource are consts, so redirect solves the problem
+                  // of incorrect names when in bch || slp.
+                  return navigation.navigate({
+                    routeName: "SendStack",
+                    key: Math.random() * 10000,
```

Comment:
What's the purpose of `key` in this navigation call?

Reply:
needed to redirect onto the same stack.

a non random key might result in the screen not changing.